src/core/server/services/users/users.ts

```diff
@@ -238,6 +248,23 @@ export async function create(
 
   const user = await createUser(mongo, tenant.id, input, now);
 
+  // Check the new user's email address against emailDomain configurations
+  // to see if they should be set to banned or always pre-moderated
+  if (tenant.emailDomainModeration) {
+    const newUserEmailDomainModeration = checkForNewUserEmailDomainModeration(
+      user,
+      tenant.emailDomainModeration
+    );
+    if (newUserEmailDomainModeration) {
+      if (newUserEmailDomainModeration === NEW_USER_MODERATION.BAN) {
+        await banUser(mongo, tenant.id, user.id);
```

Comment:
Using `banUser` here instead of `ban` since we don't need to check banner roles, existing bans/premod/suspension/warnings, reject existing comments, or send an email in this new account case. (Similar reasoning for why using `premodUser` over `premod` below).